pkg/render/logstorage/esgateway/esgateway.go

```diff
@@ -0,0 +1,352 @@
+	}
+	if e.installation.CertificateManagement != nil {
+		e.csrImage, err = render.ResolveCSRInitImage(e.installation, is)
+		if err != nil {
+			errMsgs = append(errMsgs, err.Error())
+		}
+	}
+	if len(errMsgs) != 0 {
+		return fmt.Errorf(strings.Join(errMsgs, ","))
+	}
+	return nil
+}
+
+func (e *esGateway) Objects() (toCreate, toDelete []client.Object) {
+	toCreate = append(toCreate, e.esGatewaySecrets()...)
+	toCreate = append(toCreate, e.esGatewayElasticsearchService())
+	toCreate = append(toCreate, e.esGatewayKibanaExternalService())
+	toCreate = append(toCreate, e.esGatewayKibanaService())
+	toCreate = append(toCreate, e.esGatewayRole())
+	toCreate = append(toCreate, e.esGatewayRoleBinding())
+	toCreate = append(toCreate, e.esGatewayServiceAccount())
+	toCreate = append(toCreate, e.esGatewayDeployment())
+	return toCreate, toDelete
+}
+
+func (e *esGateway) Ready() bool {
+	return true
+}
+
+func (e *esGateway) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (e esGateway) esGatewayRole() *rbacv1.Role {
+	return &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{Kind: "Role", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      RoleName,
+			Namespace: render.ElasticsearchNamespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups:     []string{""},
+				Resources:     []string{"secrets"},
+				ResourceNames: []string{},
+				Verbs:         []string{"get"},
+			},
+		},
+	}
+}
+
+func (e esGateway) esGatewayRoleBinding() *rbacv1.RoleBinding {
+	return &rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "RoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      RoleName,
+			Namespace: render.ElasticsearchNamespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     RoleName,
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      ServiceAccountName,
+				Namespace: render.ElasticsearchNamespace,
+			},
+		},
+	}
+}
+
+func (e esGateway) esGatewayDeployment() *appsv1.Deployment {
+	replicas := int32(2)
+
+	envVars := []corev1.EnvVar{
+		{Name: "ES_GATEWAY_LOG_LEVEL", Value: "INFO"},
+		{Name: "ES_GATEWAY_ELASTIC_ENDPOINT", Value: ElasticsearchHTTPSEndpoint},
+		{Name: "ES_GATEWAY_KIBANA_ENDPOINT", Value: KibanaHTTPSEndpoint},
+		{Name: "ES_GATEWAY_HTTPS_CERT", Value: "/certs/https/tls.crt"},
+		{Name: "ES_GATEWAY_HTTPS_KEY", Value: "/certs/https/tls.key"},
+		{
+			Name:      "ES_GATEWAY_ELASTIC_USERNAME",
+			ValueFrom: secret.GetEnvVarSource(ElasticsearchUserSecret, "username", false),
+		},
+		{
+			Name:      "ES_GATEWAY_ELASTIC_PASSWORD",
+			ValueFrom: secret.GetEnvVarSource(ElasticsearchUserSecret, "password", false),
+		},
+	}
+
+	volumes := []corev1.Volume{
+		{
+			Name: VolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: render.TigeraElasticsearchCertSecret,
```

Comment:
If I understand well, this is the secret containing the key and cert for encrypting traffic. I believe you should still add an annotation for this secret to the spec.